### PR TITLE
properly handle errors when reload vcl

### PR DESCRIPTION
**Pull request by awmackowiak**

### CODE REVIEW

**Comment by szymi-** [comment]
+1

File: vaas-app/src/vaas/cluster/cluster.py [hunk 1]
```diff
@@ -205,30 +210,16 @@ def use_vcl_list(self, vcl_name, vcl_loaded_list):
                     result = False
                     self.logger.error("Cannot use vcl [%s] for server %s", vcl_name, server.ip)
             for vcl, loader, server in vcl_loaded_list:
-                executor.submit(loader.discard_unused_vcls())
+                self._discard_unused_vcls(server, loader, executor, discard_results)
+            for server, status in discard_results:
+                discard_single_result = status.result()
```
Comment (ziollek) [inline vaas-app/src/vaas/cluster/cluster.py]: why are you introducing new variable (only for one comparision) ?

File: vaas-app/src/vaas/cluster/cluster.py [hunk 2]
```diff
@@ -154,23 +151,23 @@ def _append_vcl(self, vcl, server, future_results, executor):
...
+        return discard_results
+
     def load_vcl_list(self, vcl_list):
         to_use = []
         start = time.time()
         aggregated_result = True
         with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
             future_results = []
-
+            discard_results = []
```
Comment (ziollek) [inline vaas-app/src/vaas/cluster/cluster.py]: you can narrow scope of this variable (by moving it to except block)

File: vaas-app/src/vaas/vcl/loader.py [hunk 3]
```diff
@@ -74,6 +74,7 @@ def discard_unused_vcls(self):
                     self.logger.warning("VCL %s not discarded." % vcl)
             except AssertionError:
                 self.logger.warning("VCL %s not discarded." % vcl)
+                return_value = max(return_value, VclStatus.ERROR.value)
```
Comment (ziollek) [inline vaas-app/src/vaas/vcl/loader.py]: is this value using for something ?

Comment (awmackowiak) [inline vaas-app/src/vaas/vcl/loader.py]: Yes see above. 
The result_value i for return VclStatus.

**Pull request merged**